orchagent/orchagent_restart_check.cpp

```diff
@@ -0,0 +1,145 @@
+#include <iostream>
+#include <sstream>
+
+#include <unistd.h>
+#include <getopt.h>
+
+#include "notificationproducer.h"
+#include "notificationconsumer.h"
+#include "select.h"
+#include "logger.h"
+
+
+void printUsage()
+{
+    SWSS_LOG_ENTER();
+
+    std::cout << "Usage: orchagent_restart_check [-s] " << std::endl;
+    std::cout << "    -n --noFreeze" << std::endl;
+    std::cout << "        Don't freeze orchagent even if check succeeded" << std::endl;
+    std::cout << "    -s --skipPendingTaskCheck" << std::endl;
+    std::cout << "        Skip pending task dependency check for orchagent" << std::endl;
+    std::cout << "    -w --waitTime" << std::endl;
+    std::cout << "        Wait time for response from orchagent, in milliseconds. Default value: 1000" << std::endl;
+    std::cout << "    -h --help:" << std::endl;
+    std::cout << "        Print out this message" << std::endl;
+}
+
+
+/*
+ * Before stopping orchagent for warm restart, basic state check is preferred to
+ * ensure orchagent is not in transient state, so a deterministic state may be restored after restart.
+ *
+ * Here is to implement orchagent_restart_check binary which may talk to orchagent and
+ * ask it to do self-check, return "READY " signal and freeze if everything is ok,
+ * otherwise "NOT_READY" signal should be returned.
+ *
+ * Optionally:
+ *            if --noFreeze option is provided, orchagent won't freeze.
+ *            if --skipPendingTaskCheck option is provided, orchagent won't use
+ *                 whether there is pending task existing as state check criterion.
+ */
+int main(int argc, char **argv)
+{
+    swss::Logger::getInstance().setMinPrio(swss::Logger::SWSS_INFO);
+    SWSS_LOG_ENTER();
+
+    std::string skipPendingTaskCheck = "fasle";
```

Comment:
Could you please fix the typo?

Reply:
Ok.   Fortunately it is not causing problem due to value "true" is checked 
 https://github.com/Azure/sonic-swss/blob/master/orchagent/switchorch.cpp#L179